sympy/simplify/hyperexpand.py

```diff
@@ -1971,6 +1971,12 @@ def carryout_plan(f, ops):
         res = r[0].subs(z0, z)
         if rewrite:
             res = res.rewrite(rewrite)
+        if res is S.NaN:
+            # Can happen when the formula results in an indeterminate form like (log(0) - log(0)),
+            # which often happens when |z| == 1
+            # Try taking it as a limit
+            res = r[0].rewrite('nonrepsmall')
```

Comment:
Taking the limit is a good idea, but that that should probably be from the left only in case `rewrite` is `'nonrepsmall'`. Otherwise the result could be wrong.